.gitlab/install_datadog_deps.rb

```diff
@@ -99,6 +106,14 @@
   end
 end
 
+raise "#{gem_version_mapping.keys.join(',')} are not installed." if gem_version_mapping.any?
+
+datadog_gem_path = versioned_path.join("gems/datadog-#{ENV.fetch('RUBY_PACKAGE_VERSION')}")
+libdatadog_so_file = "libdatadog_api.#{RUBY_VERSION[/\d+.\d+/]}_#{RUBY_PLATFORM}.so"
+unless File.exist?("#{datadog_gem_path}/lib/#{libdatadog_so_file}")
+  raise "Missing #{libdatadog_so_file} in #{datadog_gem_path}."
+end
```

Comment:
You can also get this check by setting the `DD_FAIL_INSTALL_IF_MISSING_EXTENSION` env variable to `"true"` when calling `gem install` :)